core/tests/test_utils.py

```diff
@@ -405,6 +405,55 @@ def search(
             return result_docs[offset:], resulting_offset
 
 
+class AuthServicesStub(python_utils.OBJECT):
```

Comment:
I'm a bit confused by the need for this. Why do we need a stub? We should only be stubbing out the Firebase APIs (and even that might not be necessary if they work fine in dev mode).

The problem with the current stub is that it also stubs out our datastore calls which is not what we want.

Also, I know there's a stub for search services. Note that this is temporary and we'll likely remove it once everything has been moved to elasticsearch. The reason it's currently needed is because the API for the App Engine search and elasticsearch are completely different, so I need to wait for the "dev mode" PR to go in before clearing that stub out (it's tracked in #11314). Just thought I'd mention it in case you were basing your implementation on that.

Reply:
Hmm yes I was, but I also wanted to avoid making calls to the Firebase SDK.

Do you think it'd be better to have the platform layer be responsible for stubbing rather than test_utils? Maybe something like `auth_services.install_auth_services_stub`?

Reply:
Is the Firebase SDK available in test mode? If so, maybe you can use it (or whatever we would be using in dev mode). If not, then just stub out the functions on firebase_admin (i.e. specifically any functions defined on firebase_admin etc.) and not the entirety of auth_services.

Let's keep test_utils responsible for the stubs since these are only run in the test environment. (But try to keep the stubs as minimal as possible.)

Reply:
Hmm OK, in that case let's cross this bridge when we get to it in the ultimate migration. I'll remove it in this PR.